FreeRTOS/FreeRTOS-Kernel
# Move cmake compile options to the example project

Description
-----------
As [FreeRTOS Forum post](https://forums.freertos.org/t/cmake-default-compiler-options-and-microchip-xc16-compatibility/18361) mentioned, some compiler doesn't support some compile options. 

This PR moves the compile options to the example project and adds this project to the PR checks. This ensures that the FreeRTOS-Kernel code does not generate any warnings for GCC with these options without enforcing them on all the users. 

Test Steps
-----------
1. goto examples/cmake_example
2. cmake -S . -Bbuild -DFREERTOS_KERNEL_TOOLCHAIN=GCC
3. cmake --build build

Checklist:
----------

Related Issue
-----------
[CMake default compiler options and Microchip XC16 compatibility](https://forums.freertos.org/t/cmake-default-compiler-options-and-microchip-xc16-compatibility/18361)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
